src/main/java/org/jabref/gui/ai/components/aichat/AiChatComponent.java

```diff
@@ -1,317 +1,263 @@
-            // 1) either to NEW_NON_EXISTENT_MESSAGE,
-            // 2) or to a new history entry.
-            if (newValue.intValue() != NEW_NON_EXISTENT_MESSAGE && showingHistoryMessage.get()) {
-                if (userPromptTextArea.getCaretPosition() == 0 || !userPromptTextArea.getText().contains("\n")) {
-                    // If there are new lines in the prompt, then it is ambiguous whether the user tries to scroll up or down in history or editing lines in the current prompt.
-                    // The easy way to get rid of this ambiguity is to disallow scrolling when there are new lines in the prompt.
-                    // But the exception to this situation is when the caret position is at the beginning of the prompt.
-                    getReversedUserMessagesStream()
-                            .skip(newValue.intValue())
-                            .findFirst()
-                            .ifPresent(userMessage ->
-                                    userPromptTextArea.setText(userMessage.singleText()));
-                }
-            } else {
-                // When currentUserMessageScroll is set to NEW_NON_EXISTENT_MESSAGE, then we should:
-                // 1) either clear the prompt, if user scrolls down the most recent history entry.
-                // 2) do nothing, if user starts to edit the history entry.
-                // We distinguish these two cases by checking showingHistoryMessage, which is true for -1 message, and false for others.
-                if (showingHistoryMessage.get()) {
-                    userPromptTextArea.setText("");
-                }
-            }
-        });
+        ListUtil.getLinkedFiles(entries).forEach(file ->
+                aiService.getIngestionService().ingest(file, bibDatabaseContext).state().addListener((obs) -> updateNotifications()));
 
-        scrollPane.needsLayoutProperty().addListener((observable, oldValue, newValue) -> {
-            if (!newValue) {
-                if (blockScroll.get() == 0) {
-                    scrollPane.setVvalue(1.0);
-                } else {
-                    blockScroll.set(blockScroll.get() - 1);
-                }
-            }
-        });
-
-        chatVBox
-                .getChildren()
-                .addAll(aiChatLogic.getChatHistory()
-                                   .getMessages()
-                                   .stream()
-                                   .map(message -> new ChatMessageComponent(message, this::deleteMessage))
-                                   .toList()
-                );
+        updateNotifications();
+    }
 
+    private void initializeNotice() {
         String newNotice = noticeText
                 .getText()
-                .replaceAll("%0", aiPreferences.getAiProvider().getLabel() + " " + aiPreferences.getSelectedChatModel());
+                .replaceAll("%0", aiService.getPreferences().getAiProvider().getLabel() + " " + aiService.getPreferences().getSelectedChatModel());
```

Comment:
I think, the preferences should be "global" -> meaning injected using the constructor. An indirection through the service feels strange. At least, it is consistent to the other code in JabRef.